### PR TITLE
[ADD] stock_restrict_lot_update

### PR DESCRIPTION
**Pull request by HekkiMelody**
This module builds on `stock_restrict_lot` and exposes the field
`restrict_lot_id` allowing the user to update or even remove
the lot restriction if the product does not fall within a domain.

The domain can be updated in the settings of each individual company.

TODO:
- [x] Tests
- [x] documentation

### CODE REVIEW

**Review by ilyasProgrammer — APPROVED** [review]
LGTM
picky: missing copyright in files

**Comment by HekkiMelody** [comment]
> LGTM
> picky: missing copyright in files

Added copyright for the sake of having the PR merged, though I'm not sure it's strictly necessary.

File: stock_restrict_lot_update/tests/test_restrict_lot_update.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 Ooops404
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import SavepointCase
+
+
+class TestRestrictLotUpdate(SavepointCase):
```
Comment (geomer198) [inline stock_restrict_lot_update/tests/test_restrict_lot_update.py:7]: Could you add docstring to each test case?

File: stock_restrict_lot_update/models/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                move.sudo().company_id.enforce_lot_restriction_product_domain
+            )
+            # if product is not in the enforcement domain, user can change the lot
+            move.change_restrict_lot = not move.product_id.filtered_domain(
+                product_domain
+            )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "restrict_lot_id" in vals:
```
Comment (geomer198) [inline stock_restrict_lot_update/models/stock_move.py:33]: Could you add a comment about what this code does?

**Review by geomer198 — COMMENTED** [review]
I think this is optional but code will look better.

File: stock_restrict_lot_update/tests/test_restrict_lot_update.py (hunk 1 above)
Comment (HekkiMelody) [reply stock_restrict_lot_update/tests/test_restrict_lot_update.py:7]: Done!

File: stock_restrict_lot_update/models/stock_move.py (hunk 2 above)
Comment (HekkiMelody) [reply stock_restrict_lot_update/models/stock_move.py:33]: Done!

**Review by geomer198 — APPROVED** [review]
LGTM!

File: stock_restrict_lot_update/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1,5 @@
+* Ooops404
+
```
Comment (francesco-ooops) [inline stock_restrict_lot_update/readme/CONTRIBUTORS.rst:2]: * Francesco Foresti <francesco.foresti@ooops404.com>

**Review by francesco-ooops — APPROVED** [review]
Functional ok, we got USAGE file too

@pedrobaeza good for you?

**Comment by pedrobaeza** [comment]
I think in this case a CONTEXT file explaining why do you want to do this update is interesting.

**Comment by HekkiMelody** [comment]
> I think in this case a CONTEXT file explaining why do you want to do this update is interesting.

I've added a CONTEXT file.

**Comment by pedrobaeza** [comment]
Thanks:

/ocabot merge nobump

**Pull request merged**